packages/components/src/core/DropdownMenu/index.tsx

```diff
@@ -36,66 +35,65 @@ interface ExtraDropdownMenuProps extends StyleProps {
   title?: string;
   label?: string;
   anchorEl: HTMLElement | null;
-  PopperComponent?: typeof StyledPopper | RenderFunctionType;
+  PopperComponent?: React.JSXElementConstructor<PopperProps>;
   PopperPlacement?: "bottom-start" | "top-start" | "bottom-end" | "top-end";
-  PaperComponent?: typeof StyledPaper | RenderFunctionType;
   children?: JSX.Element | null;
-  onClickAway: (event: MouseEvent | TouchEvent) => void;
+  onClickAway?: (event: MouseEvent | TouchEvent) => void;
   ClickAwayListenerProps?: Partial<MUIClickAwayListenerProps>;
+  width?: number;
 }
 
 type CustomAutocompleteProps<
   T,
-  Multiple extends boolean | undefined = undefined,
-  DisableClearable extends boolean | undefined = undefined,
-  FreeSolo extends boolean | undefined = undefined
+  Multiple extends boolean | undefined,
+  DisableClearable extends boolean | undefined,
+  FreeSolo extends boolean | undefined
 > = Omit<
   AutocompleteProps<T, Multiple, DisableClearable, FreeSolo>,
   "renderInput" | "nonce" | "rev" | "rel" | "autoFocus" | "content"
 >;
 
 export type DropdownMenuProps<
   T,
-  Multiple extends boolean | undefined = undefined,
-  DisableClearable extends boolean | undefined = undefined,
-  FreeSolo extends boolean | undefined = undefined
+  Multiple extends boolean | undefined,
+  DisableClearable extends boolean | undefined,
+  FreeSolo extends boolean | undefined
 > = CustomAutocompleteProps<T, Multiple, DisableClearable, FreeSolo> &
   ExtraDropdownMenuProps;
 
 const DropdownMenu = <
-  T extends DefaultDropdownMenuOption,
-  Multiple extends boolean | undefined = undefined,
-  DisableClearable extends boolean | undefined = undefined,
-  FreeSolo extends boolean | undefined = undefined
+  T extends DefaultAutocompleteOption,
+  Multiple extends boolean | undefined,
+  DisableClearable extends boolean | undefined,
+  FreeSolo extends boolean | undefined
 >(
   props: DropdownMenuProps<T, Multiple, DisableClearable, FreeSolo>
 ): JSX.Element => {
+  const theme: SDSTheme = useTheme();
+
   const {
     anchorEl,
     id,
-    InputBaseProps = {
-      autoFocus: true,
-      sx: {
-        padding: "8px",
-      },
-    },
+    InputBaseProps,
     open = false,
-    PaperComponent = StyledPaper,
     PopperComponent = StyledPopper,
     PopperPlacement = "bottom-start",
     PopperBaseProps = {},
     search = false,
     title,
     label = "Search",
     children,
-    onClickAway,
+    options,
+    onClickAway = noop,
     ClickAwayListenerProps,
+    width = 160,
```

Comment:
can we extract this magic number out as a constant and add a comment to explain the rationale behind this 160 value please? And we can then reuse the constant in `width < 160` below!

Reply:
Oh, nice catch 😱 

```javascript
// (masoudmanson): Represents the minimum width defined by design specifications
// for the dropdownMenu's popper component
export const MINIMUM_DROPDOWN_MENU_POPPER_WIDTH = 160;
```